src/app-layout/utils/use-drawers.ts

```diff
@@ -50,12 +52,21 @@ function useRuntimeDrawers(
 ) {
   const [runtimeDrawers, setRuntimeDrawers] = useState<DrawersLayout>({ before: [], after: [] });
 
+  const prevDrawersRef = useRef<Array<RuntimeDrawerConfig>>([]);
+
   useEffect(() => {
     if (disableRuntimeDrawers) {
       return;
     }
     const unsubscribe = awsuiPluginsInternal.appLayout.onDrawersRegistered(drawers => {
       setRuntimeDrawers(convertRuntimeDrawers(drawers));
+      if (!prevDrawersRef.current || !prevDrawersRef.current.find(drawer => drawer.defaultActive)) {
```

Comment:
If the old drawer is closed, shall the new defaultActive drawer open? Maybe check `activeDrawerId` so we wont open a new drawer when there is one open already?

Reply:
Good point, updated the implementation.

The new logic is also much simpler "defaultActive only works when there was no any open drawer before. If there was an open drawer (by a user or from another instance), new drawer won't open"